### ISSUE TITLE
Cannot convert NULL to a non-nullable type / LowCardinality(Nullable(String))

### ISSUE THREAD

**Issue by champtar** [yandex/ClickHouse#3066]
Hello ClickHouse team,

Here some failing requests:
```
SELECT CAST(NULL, 'LowCardinality(Nullable(Int8))');
SELECT CAST(NULL, 'NullableWithDictionary(Int8)');
CREATE TABLE test.lc_null_int8_defnull_0 (val LowCardinality(Nullable(Int8)) DEFAULT NULL) ENGINE = Memory;
CREATE TABLE test.lc_null_int8_defnull_1 (val NullableWithDictionary(Int8) DEFAULT NULL) ENGINE = Memory;

=> DB::Exception: Cannot convert NULL to a non-nullable type.
```

**Comment by champtar** [comment]
issue present on 18.12.6

**Comment by champtar** [comment]
Fixed in 18.14

**Comment by Daletxt** [comment]
I can use 
"""INSERT INTO table_name (contain_NULL_value_field, other_filed) SELECT ifNull('contain_NULL_value_field', 'your_target_replace_value') """
to solve this problem.
